### PR TITLE
HW06 Gerasev Georgii

### PR DESCRIPTION
**Pull request by VenInf**

### CODE REVIEW

File: monads/src/Either.hs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    (MyRight u) <*> pure y = (MyRight u) <*> (MyRight y) = MyRight (u y) = ($ y) <$> (MyRight u) = MyRight ($ y) <*> (MyRight u) 
+-}
+
+-- Implement the instance and prove the laws
+instance Monad (MyEither a) where
+  (>>=) :: MyEither a b1 -> (b1 -> MyEither a b2) -> MyEither a b2
+  MyLeft l >>= _ = MyLeft l
+  MyRight r >>= k = k r
+  return :: b -> MyEither a b 
+  return = pure
```
Comment (kajigor) [inline monads/src/Either.hs:56]: No need to implement `return`, it has this exact implementation by default.

File: monads/src/Either.hs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  fmap f (MyRight x) = MyRight (f x)
+
+{-
+  1) fmap id = id
+  fmap id (MyLeft x) = (MyLeft x)
+  fmap id (MyRight x) = MyRight (id x) =  MyRight x  
+  
+  2) fmap (f . g)  ==  fmap f . fmap g
+  fmap (f . g) (MyLeft x) = MyLeft x = fmap f (MyLeft x) = fmap f . fmap g (MyLeft x)
+  fmap (f . g) (MyRight x) = MyRight (f.g) x = fmap f (MyRight g x) = fmap f . fmap g (MyRight x)
```
Comment (kajigor) [inline monads/src/Either.hs:21]: The first equality is incorrect.

**Review by kajigor — CHANGES_REQUESTED** [review]

**Comment by kajigor** [comment]
1.1 ok, the rest is not implemented

File: monads/src/Either.hs (hunk 2 above)
Comment (VenInf) [reply monads/src/Either.hs]: Tried to fix it.
Not quite sure where the error is.

**Comment by kajigor** [comment]
ok

**Pull request closed without merging**